spec/index.bs

```diff
@@ -6590,9 +6590,11 @@ pipeline, and have the following members:
                 [=Device timeline=] |initialization steps|:
 
                 1. Let |limits| be |this|.{{GPUObjectBase/[[device]]}}.{{device/[[limits]]}}.
-                1. Let |bindGroupLayouts| be a copy of |descriptor|.{{GPUPipelineLayoutDescriptor/bindGroupLayouts}}.
-                1. For each |i| in the [=list/get the indices|indices=] of |bindGroupLayouts|:
-                    1. If |bindGroupLayouts|[|i|] is not `null` and
+                1. Let |bindGroupLayouts| be a [=list=] of `null` {{GPUBindGroupLayout}}s with [=list/size=]
+                     equal to |limits|.{{supported limits/maxBindGroups}}.
+                1. For each |i| in the [=list/get the indices|indices=] of |descriptor|.{{GPUPipelineLayoutDescriptor/bindGroupLayouts}}:
+                    1. Set |bindGroupLayouts|[|i|] to ||descriptor|.{{GPUPipelineLayoutDescriptor/bindGroupLayouts}}|[|i|].
+                    1. If |bindGroupLayouts|[|i|] is `undefined` or |bindGroupLayouts|[|i|] is not `null` and
```

Comment:
nit: I don't think the check for `undefined` is needed, because bindGroupLayouts can't contain undefined (and `i` is known to be in-bounds - this is ["asserted" by the indexing operation](https://infra.spec.whatwg.org/#lists:~:text=An%20indexing%20syntax%20can%20be%20used%20by%20providing%20a%20zero%2Dbased%20index%20into%20a%20list%20inside%20square%20brackets.%20The%20index%20cannot%20be%20out%2Dof%2Dbounds%2C%20except%20when%20used%20with%20exists.) anyway).
```suggestion
                    1. If |bindGroupLayouts|[|i|] is not `null` and
```